be/src/service/backend_service.cpp

```diff
@@ -119,7 +119,9 @@ void BackendService::transmit_data(TTransmitDataResult& return_val, const TTrans
 
 void BackendService::fetch_data(TFetchDataResult& return_val, const TFetchDataParams& params) {
     // maybe hang in this function
-    Status status = _exec_env->result_mgr()->fetch_data(params.fragment_instance_id, &return_val);
+    auto result = std::make_unique<TFetchDataResult>();
```

Comment:
Why make a new TFetchDataResult?

Reply:
there is a swap operation in `fetch_data`, the pointer might be changed.